driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -243,6 +243,38 @@ public static Bson lookup(final String from, final String localField, final Stri
                                                    .append("as", new BsonString(as)));
     }
 
+    /**
+     * Creates a $lookup pipeline stage for the specified filter
+     *
+     * @param from          the name of the collection in the same database to perform the join with.
+     * @param pipeline      the pipeline to run on the joined collection.
+     * @param as            the name of the new array field to add to the input documents.
+     * @return the $lookup pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/lookup/ $lookup
+     * @mongodb.server.release 3.6
+     * @since 3.7
+     *
+     */
+    public static Bson lookup(final String from, final List<? extends Bson> pipeline, final String as) {
```

Comment:
I'd like to see a unit and functional test for this overload.  Currently it's untested, but I want to make sure we catch any issues when `let` is null.